pkg/asset/installconfig/azure/validation.go

```diff
@@ -0,0 +1,71 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	aznetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+
+	"github.com/openshift/installer/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// Validate executes platform-specific validation.
+func Validate(client API, ic *types.InstallConfig) error {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, validateNetworks(client, ic, field.NewPath("platform").Child("azure"))...)
+	return allErrs.ToAggregate()
+}
+
+// validateNetworks checks that the user-provided VNet is in the project and the provided subnets are valid.
+func validateNetworks(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.Azure.VirtualNetwork != "" {
+		_, err := client.GetVirtualNetwork(context.TODO(), ic.Azure.NetworkResourceGroupName, ic.Azure.VirtualNetwork)
+		if err != nil {
+			return append(allErrs, field.Invalid(fieldPath.Child("virtualNetwork"), ic.Azure.VirtualNetwork, err.Error()))
+		}
+
+		computeSubnet, err := client.GetSubnet(context.TODO(), ic.Azure.NetworkResourceGroupName, ic.Azure.VirtualNetwork, ic.Azure.ComputeSubnet)
+		if err != nil {
+			return append(allErrs, field.Invalid(fieldPath.Child("computeSubnet"), ic.Azure.VirtualNetwork, "failed to retrieve compute subnet"))
+		}
+
+		controlPlaneSubnet, err := client.GetSubnet(context.TODO(), ic.Azure.NetworkResourceGroupName, ic.Azure.VirtualNetwork, ic.Azure.ControlPlaneSubnet)
+		if err != nil {
+			return append(allErrs, field.Invalid(fieldPath.Child("controlPlaneSubnet"), ic.Azure.VirtualNetwork, "failed to retrieve control plane subnet"))
+		}
+
+		computeSubnetErrs := append(allErrs, validateSubnet(client, ic, fieldPath.Child("computeSubnet"), computeSubnet, ic.Azure.ComputeSubnet)...)
+		if computeSubnetErrs == nil {
```

Comment:
it will never be nil right, seehttps://github.com/openshift/installer/pull/2613/commits/96d7f5467119ff6465105737bbdebfeb5b70d14f#diff-4ead0ce7a6dc0f22e8747033070afe8aR57 it creates `ErrorList` first thing.